internal/proto/stdout_stderr_writer.go

```diff
@@ -0,0 +1,50 @@
+package proto
+
+import (
+	"io"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"os"
+)
+
+type StdoutStderrWriter struct {
+	stdout io.Writer
+	stderr io.Writer
+}
+
+func printOutput(output Output) error {
+	errs := make([]error, 0)
+
+	stdout := output.GetStdout()
+	if stdout != nil {
+		_, err := os.Stdout.Write(stdout)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	stderr := output.GetStderr()
+	if stderr != nil {
+		_, err := os.Stderr.Write(stderr)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return errors.NewAggregate(errs)
```

Comment:
this seems very specific to certain kubernetes patterns where they really do need an aggregate error. I don't think this is a good pattern to copy here.

it's generally easier to bail out at the first error, and easier to handle upstream (rather than make the caller figure out what to do with aggregates)